sounddevice.py

```diff
@@ -2265,10 +2265,66 @@ def reset(self):
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
-
+    # A simple enumeration of the Portaudio error codes
+    PaErrorCode = type('PaErrorCode', (), _ffi.typeof('PaErrorCode').relements)
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    msg : optional
+        User error message.
+    errcode : int or None, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
+
+    Notes
+    -----
+    - Some PortAudioError's do not have an associated PaErrorCode. In
+      such a case `paerrcode` will be None.
+
+    - The `hostapi`, `hosterrcode`, and `hosterrmsg` attributes will be
+      None except for the case when `paerrcode` ==
+      `paUnanticipatedHostError`.
+    
+    Attributes
+    ----------
+    paerrcode : int or None
+        The PaErrorCode integer value or None if there is no associated
+        error code.
+    hostapi : str or None
```

Comment:
I think it would make more sense to expose the host API number instead of its name.
`query_hostapis()` expects the number, the name doesn't really help that much, programmatically.
The error message should of course keep using the name.